synapse/handlers/federation.py

```diff
@@ -608,6 +612,14 @@ async def do_invite_join(
                     room_id,
                 )
                 raise LimitExceededError(msg=e.msg, errcode=e.errcode, retry_after_ms=0)
+            else:
+                # Record the join event id for future use (when we finish the full
+                # join). We have to do this after persisting the event to keep foreign
+                # key constraints intact.
+                if ret.partial_state:
+                    await self.store.write_partial_state_rooms_join_event_id(
+                        room_id, event.event_id
+                    )
```

Comment:
Why do we do this in an `else` block rather than in the `try` or afterwards?

Reply:
My thinking was that

- If `process_remote_join`  fails then the event may not be persisted
- keeping the try block small helps clarify what the `PartialStateConflictError` is catching

Happy to just bung it at the bottom of the `try` though if you prefer.